src/index.ts

```diff
@@ -0,0 +1,104 @@
+import { State } from './types';
+
+const defaultState: State = {
+  shipmentId: 0,
+  toAddress: '',
+  fromAddress: '',
+  toZipCode: '',
+  fromZipCode: '',
+  weight: 0,
+  marks: [],
+};
+
+class Gui {
+  state: State;
+  constructor() {
+    this.state = defaultState;
+  }
+
+  // on(eventType: string): void;
+
+  // trigger(eventType: string, state: Shipment): any {}
+
+  getState(): State {
+    return this.state;
+  }
+}
+
+class Client {
+  gui;
+
+  constructor(gui: Gui) {
+    this.gui = gui;
+  }
+
+  onShip(shipment: Shipment) {}
+}
+
+class Shipment {
+  state: State;
+
+  constructor(state: State) {
+    this.state = state;
+  }
+
+  getShipmentId(): number {
+    return this.state.shipmentId;
+  }
+
+  ship(state: State) {
+    if (state.weight < 15) {
+      return this.Letter();
+    } else if (state.weight <= 160) {
+      return this.Package();
+    } else {
+      return this.Oversize();
+    }
+  }
+
+  Letter() {}
```

Comment:
Types of shipment should be implemented using classes and have logic in them.

Reply:
need help :)